src/frontends/paddle/src/op/gather_nd.cpp

```diff
@@ -0,0 +1,26 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "default_opset.hpp"
+#include "openvino/frontend/paddle/node_context.hpp"
+
+namespace ov {
+namespace frontend {
+namespace paddle {
+namespace op {
+NamedOutputs gather_nd(const NodeContext& node) {
+    const auto data_node = node.get_input("X");
+    const auto index_node = node.get_input("Index");
+    auto shape = index_node.get_partial_shape();
+    for (const auto& dim : shape) {
+        if (dim.is_static() && dim.get_length() == 0)
+            PADDLE_OP_CHECK(node, false, "zero dimension is not allowed for gather_nd Index");
```

Comment:
As I see from the specification zero index is supported by gather ND op:
https://docs.openvino.ai/latest/openvino_docs_ops_movement_GatherND_8.html#doxid-openvino-docs-ops-movement-gather-n-d-8

Reply:
Hi, @ilyachur : thanks for taking your time to help a look at this PR. the 'indices' input value of OV 'gather_nd' can be '0', but as its specification describes 'A tensor of a rank not less than 1'. the 'indices' input rank should not be '0'(@AndPuQing  has created an commented test case 'gather_nd_empty' to cover this). 
Maybe the 'message info' the this 'PADDLE_OP_CHECK' is a little bit Confusing. Do you think better we change it from 'zero dimension is not allowed for gather_nd Index' to 'zero 'indices' input rank is not allowed for gather_nd'?

Reply:
Yes I think it will be more aligned with the specification